test/runtests.jl

```diff
@@ -80,8 +80,7 @@ abstract type AbstractTable end
         Tables.rowaccess(::Type{AbstractTable}) ||
             Tables.columnaccess(::Type{AbstractTable})
     if Tables.isrowtable(AbstractTable) || Tables.rowaccess(AbstractTable)
-        Tables.rows(::AbstractTable)::Iterable
-        RT = Interfaces.@returntype Tables.rows(::AbstractTable)
+        Tables.rows(::AbstractTable) -> RT::Iterable
```

Comment:
in case you were wondering "hmm, what happens if someone tries to spell this `Tables.rows(::AbstractTable)::Iterable -> RT`?"

this is what happens:
```jl
ERROR: LoadError: LoadError: AssertionError: x.head === :call
Stacktrace:
  [1] methodparts(T::Symbol, x::Expr)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:65
  [2] toimplements!(T::Symbol, arg::Expr, shouldthrow::Bool)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:102
  [3] (::Interfaces.var"#1#2"{Symbol, Bool})(x::Expr)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:148
  [4] map!(f::Interfaces.var"#1#2"{Symbol, Bool}, dest::Vector{Any}, A::Vector{Any})
    @ Base ./abstractarray.jl:2287
  [5] toimplements!(T::Symbol, arg::Expr, shouldthrow::Bool)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:148
  [6] toimplements!(T::Symbol, arg::Expr)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:90
  [7] toimplements!(T::Symbol, arg::Expr, shouldthrow::Bool)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:121
  [8] (::Interfaces.var"#1#2"{Symbol, Bool})(x::Expr)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:148
  [9] map!(f::Interfaces.var"#1#2"{Symbol, Bool}, dest::Vector{Any}, A::Vector{Any})
    @ Base ./abstractarray.jl:2287
 [10] toimplements!(T::Symbol, arg::Expr, shouldthrow::Bool)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:148
 [11] toimplements!(T::Symbol, arg::Expr)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:90
 [12] var"@interface"(__source__::LineNumberNode, __module__::Module, T::Any, block::Any)
    @ Interfaces ~/repos/Interfaces.jl/src/Interfaces.jl:161
```

Perhaps we want that to fail a little more gracefully?

Reply:
In my `foo() -> T || foo() -> x::T` proposal, we wouldn't allow `foo()::T` any more, so we could just throw a helpful error saying they should do `foo() -> T` instead, or do `foo() -> RT::Iterable` in the case you mentioned above.